images/cfw/usr/etc/system/webhmi/little_webhmi.py

```diff
@@ -0,0 +1,112 @@
+# aiohttp variant
+import aiohttp
+from aiohttp import web
+import asyncio
+import subprocess
+import json
+import threading
+import aiohttp_jinja2
+import jinja2
+import os
+
+# Print the current working directory for debugging
+print("Current working directory:", os.getcwd())
+
+# Global variable to hold the current job name
+current_job_name = "None"
+# Global variable to hold the latest message
+latest_message = ""
+
+async def update_current_job():
+    global current_job_name
+    while True:
+        try:
+            with open('/userdata/print_ctx.json', 'r') as file:
+                data = json.load(file)
+                subtask_name = data['subtask_name']
+                current_job_name = subtask_name
+        except Exception as e:
+            print(f"Error reading or parsing JSON file: {e}")
+        await asyncio.sleep(10)
+
+@aiohttp_jinja2.template('index.html')
+async def printer_hmi(request):
+    printer_details = {
+        'status': 'Printing',
+        'current_job': current_job_name,
+        'progress': 75,
+        'temperature': {
+            'nozzle': 215,
+            'bed': 60
+        },
+        'latest_message': latest_message,
+    }
+    return {'details': printer_details}
+
+async def home_xyz_func(request):
+    global latest_message
+    subprocess.run(["/usr/bin/home_xyz.sh"], shell=False)
+    latest_message = 'Home XYZ'
+    raise web.HTTPFound('/')
+
+async def preheat_100c(request):
+    global latest_message
+    subprocess.run(["/usr/bin/heatbed_set.sh", "-s", "100"], shell=False)
+    latest_message = 'Preheat Activated - 100c'
+    raise web.HTTPFound('/')
+
+async def preheat_0c(request):
+    global latest_message
+    subprocess.run(["/usr/bin/heatbed_set.sh", "-s", "0"], shell=False)
+    latest_message = 'Preheat Deactivated'
+    raise web.HTTPFound('/')
```

Comment:
One thing I am thinking about is that I would rather have a generic interface to the printer -- with logic run in a SPA in the browser -- than having one HTTP endpoint for each of a million little things.  I was thinking that the generic interface to the printer really wanted to be a WebSocket with generic interfaces like 'submit Gcode', or 'submit MQTT request', or 'execute DBus RPC', with all the smarts on the client side.